### PR TITLE
API 800/1000/1200 fc network support

### LINKED ISSUE #401: Travis build is failing at multiple points due to incompatibilty between the dependencies
### Scenario/Intent
Travis build is failing at multiple points due to incompatibilty between the dependencies

### Environment Details
 - **oneview Cookbook Version:** 3.1.0
 - **OneView Version:** 5.10.0
 - **chef-client Version:** 14.14.29
 - **platform:** redhat

### Steps to Reproduce
Travis build fails for ruby 2.4, 2.5 but pass for 2.3.1

### Expected Result
Should be able to pass tests for 2.4 and 2.5

### Actual Result
No Chef resources found, skipping coverage calculation...

SimpleCov failed with exit 1
/home/travis/.rvm/rubies/ruby-2.4.6/bin/ruby -I/home/travis/.rvm/gems/ruby-2.4.6/gems/rspec-core-3.9.1/lib:/home/travis/.rvm/gems/ruby-2.4.6/gems/rspec-support-3.9.2/lib /home/travis/.rvm/gems/ruby-2.4.6/gems/rspec-core-3.9.1/exe/rspec --pattern spec/\*\*/\*_spec.rb --color  failed

The command "rake test" exited with 1.

1.00s$ bundle exec codeclimate-test-reporter

I, [2020-03-06T06:36:15.860550 #16801]  INFO -- : Reporting coverage data to Code Climate.

/home/travis/.rvm/gems/ruby-2.4.6/gems/codeclimate-test-reporter-1.0.7/lib/code_climate/test_reporter/formatter.rb:54:in `block in to_payload': undefined method `coverage' for #<SimpleCov::SourceFile:0x00000000026c4e20> (NoMethodError)

Did you mean?  coverage_data

	from /home/travis/.rvm/rubies/ruby-2.4.6/lib/ruby/2.4.0/forwardable.rb:229:in `map'

	from /home/travis/.rvm/rubies/ruby-2.4.6/lib/ruby/2.4.0/forwardable.rb:229:in `map'

	from /home/travis/.rvm/gems/ruby-2.4.6/gems/codeclimate-test-reporter-1.0.7/lib/code_climate/test_reporter/formatter.rb:41:in `to_payload'

	from /home/travis/.rvm/gems/ruby-2.4.6/gems/codeclimate-test-reporter-1.0.7/lib/code_climate/test_reporter/formatter.rb:27:in `format'

	from /home/travis/.rvm/gems/ruby-2.4.6/gems/codeclimate-test-reporter-1.0.7/lib/code_climate/test_reporter.rb:19:in `run'

	from /home/travis/.rvm/gems/ruby-2.4.6/gems/codeclimate-test-reporter-1.0.7/bin/codeclimate-test-reporter:21:in `<top (required)>'

	from /home/travis/.rvm/gems/ruby-2.4.6/bin/codeclimate-test-reporter:23:in `load'

	from /home/travis/.rvm/gems/ruby-2.4.6/bin/codeclimate-test-reporter:23:in `<main>'

	from /home/travis/.rvm/gems/ruby-2.4.6/bin/ruby_executable_hooks:24:in `eval'

	from /home/travis/.rvm/gems/ruby-2.4.6/bin/ruby_executable_hooks:24:in `<main>'

The command "bundle exec codeclimate-test-reporter" exited with 1.

### PR DESCRIPTION
**Pull request by soodpr**
### Description
API 800/1000/1200 FC network support,
Removed ruby 2.2 support as OneView ruby sdk is not supporting it any  longer and added ruby 2.5 support.

### Issues Resolved
 #401 

### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [X] New functionality has been documented in the README if applicable.
  - [X] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [X] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: spec/unit/libraries/resource_provider_spec.rb [hunk 1]
```diff
@@ -93,6 +93,54 @@
       expect(r.item.class).to eq(OneviewSDK::API600::Synergy::Scope)
     end
 
+    it 'builds valid API800::C7000 resources' do
```
Comment (AnushaY1916) [inline spec/unit/libraries/resource_provider_spec.rb:96]: Why do we need to add these here?

**Review by AnushaY1916 — COMMENTED** [review]
SDK link in endpoint file pointing to OneView 4.1. We can update it to 5.

File: spec/unit/libraries/resource_provider_spec.rb (hunk 1 above)
Comment (soodpr) [reply spec/unit/libraries/resource_provider_spec.rb:96]: it is require to test whether a valid resource is created as per its API version and variant class. Its not necessary to include it but i have added just to keep this file in sync with the api's we are supporting.

**Comment by soodpr** [comment]
> 
> 
> SDK link in endpoint file pointing to OneView 4.1. We can update it to 5.

done.Thanks for pointing it out.

**Review by AnushaY1916 — APPROVED** [review]
LGTM

**Review by jyothisgm — APPROVED** [review]
LGTM

**Pull request merged**